### ISSUE TITLE
--text only seems to apply to navigation

### ISSUE THREAD

**Issue by david1602** [widget-/slack-black-theme#39]
Today Slack has been down and after refreshing the page, the `--text` part of the custom CSS wouldn't apply to the body anymore. My friend doesn't seem to have this issue, but I do on two different devices with two different operating systems, so I think it can't just be me messing something up.

Am I the only person running into this? I tested this both on mac and windows 10, and I used the actual Slack application, rather than something from the apple / windows store.

Below is a screenshot of my troubles:
![image](https://user-images.githubusercontent.com/11957414/34744289-06dfd884-f58d-11e7-9cf6-4ff5447695be.png)

Hovering works, sort of

![image](https://user-images.githubusercontent.com/11957414/34744309-1980a1b2-f58d-11e7-9514-f5a3a35db384.png)

**Comment by Arezhik** [comment]
Myself and others I work with are experiencing it too.

I added the following to the let customCustomCSS to at least make it readable @david1602 

`	div.c-message.c-message--light.c-message--hover
	{
		color: #fff !important;
		background-color: #222 !important;
	}
	
	span.c-message__body,
	a.c-message__sender_link,
	span.c-message_attachment__media_trigger.c-message_attachment__media_trigger--caption,
	div.p-message_pane__foreword__description span
	{
			color: #8f8f8f !important;
	}
	
	pre.special_formatting{
		background-color: #222 !important;
		color: #8f8f8f !important;
		border: solid;
		border-width: 1 px !important;
		
	}`

**Comment by david1602** [comment]
I tried but I failed badly trying to do the same thing, that helps a lot! I think an improvement label for the documentation would be appropriate.

**Comment by stevenlordiam** [comment]
Hi I had the same issue, where should I add your customCustomCSS to? @Arezhik

**Comment by david1602** [comment]
@stevenlordiam 
this is what the bottom of my `ssb-interop.js` file looks like:

```javascript
// First make sure the wrapper app is loaded
document.addEventListener("DOMContentLoaded", function() {

   // Then get its webviews
   let webviews = document.querySelectorAll(".TeamView webview");

   // Fetch our CSS in parallel ahead of time
   const cssPath = 'https://cdn.rawgit.com/widget-/slack-black-theme/master/custom.css';
   let cssPromise = fetch(cssPath).then(response => response.text());

   let customCustomCSS = `
   :root {
      /* Modify these to change your theme colors: */
      --primary: #09F;
      --text: #CCC;
      --background: #080808;
      --background-elevated: #222;
   }
   div.c-message.c-message--light.c-message--hover
	{
	color: #fff !important;
	background-color: #222 !important;
	}

	span.c-message__body,
	a.c-message__sender_link,
	span.c-message_attachment__media_trigger.c-message_attachment__media_trigger--caption,
	div.p-message_pane__foreword__description span
	{
			color: #afafaf !important;
	}

	pre.special_formatting{
		background-color: #222 !important;
		color: #8f8f8f !important;
		border: solid;
		border-width: 1 px !important;
		
	}
   `

   // Insert a style tag into the wrapper view
   cssPromise.then(css => {
      let s = document.createElement('style');
      s.type = 'text/css';
      s.innerHTML = css + customCustomCSS;
      document.head.appendChild(s);
   });

   // Wait for each webview to load
   webviews.forEach(webview => {
      webview.addEventListener('ipc-message', message => {
         if (message.channel == 'didFinishLoading')
            // Finally add the CSS into the webview
            cssPromise.then(css => {
               let script = `
                     let s = document.createElement('style');
                     s.type = 'text/css';
                     s.id = 'slack-custom-css';
                     s.innerHTML = \`${css + customCustomCSS}\`;
                     document.head.appendChild(s);
                     `
               webview.executeJavaScript(script);
            })
      });
   });
});
```

**Comment by stevenlordiam** [comment]
@david1602 it works for me, thank you!

**Comment by david1602** [comment]
@stevenlordiam you're welcome!

**Comment by Arezhik** [comment]
Welcome guys. I do have a my own repo based off of this CSS, you are welcome to pull from it for your cssPath variable in the same file.

`
// First make sure the wrapper app is loaded
document.addEventListener("DOMContentLoaded", function() {

   // Then get its webviews
   let webviews = document.querySelectorAll(".TeamView webview");

   // Fetch our CSS in parallel ahead of time
   const cssPath = "https://cdn.rawgit.com/Arezhik/slackCustomCSS/85e49edd/custom.css"
   let cssPromise = fetch(cssPath).then(response => response.text());

	let customCustomCSS = `
	:root{
	/*Modifythesetochangeyourthemecolors: */
	--primary: #09F;
	--text: #CCC;
	--background: #080808;
	--background-elevated: #222;
	}`

   // Insert a style tag into the wrapper view
   cssPromise.then(css => {
      let s = document.createElement('style');
      s.type = 'text/css';
      s.innerHTML = css + customCustomCSS;
      document.head.appendChild(s);
   });

   // Wait for each webview to load
   webviews.forEach(webview => {
      webview.addEventListener('ipc-message', message => {
         if (message.channel == 'didFinishLoading')
            // Finally add the CSS into the webview
            cssPromise.then(css => {
               let script = `
                     let s = document.createElement('style');
                     s.type = 'text/css';
                     s.id = 'slack-custom-css';
                     s.innerHTML = \`${css + customCustomCSS}\`;
                     document.head.appendChild(s);
                     `
               webview.executeJavaScript(script);
            })
      });
   });
});`

**Comment by david1602** [comment]
@Arezhik I appreciate the effort, but you have to fix your formatting 😄

**Comment by Arezhik** [comment]
Work in progress, the theme was broken for me to begin with unfortunately and didn’t work well with slack 3.0.0.

If you feel the need you can create an issue on my repository for it, so I can look into it.

**Comment by Zoooook** [comment]
None of those solutions worked for me, no matter whose CSS I use none of my own colors apply to anything except the far left workspace-switcher sidebar. And then I upgraded to Slack 3.0 and the whole thing is busted, back to white background. :(

**Comment by kivan-mih** [comment]
Solution by david1602 works, but with two problems:
1. Color is not applied to shared message
2. When scrolling message history some jittering occur.
I have slack 3.0.5 on linux mint 18

**Comment by Bterrell721** [comment]
Same issue as @kivan-mih.

david1602's fix does resolve the issue with impossible to read text and highlight issue, but the text color in the main message pane is reverted to grey and doesn't respect my color choices in the "root" section.

Any fix for that @david1602 ?

**Comment by Jyosua** [comment]
I'm also experiencing the same thing as @Bterrell721 as of last night.

**Comment by laurencee** [comment]
Slight modification to fix bot link text, add `span.c-message_attachment__text` to 
```css
span.c-message__body,
a.c-message__sender_link,
span.c-message_attachment__media_trigger.c-message_attachment__media_trigger--caption,
div.p-message_pane__foreword__description span
{
	color: #afafaf !important;
}
```

So you end up with 
```css
span.c-message_attachment__text,
span.c-message__body,
a.c-message__sender_link,
span.c-message_attachment__media_trigger.c-message_attachment__media_trigger--caption,
div.p-message_pane__foreword__description span
{
	color: #afafaf !important;
}
```

**Comment by me21** [comment]
Another issue is that "Add reaction" button is invisible until you hover mouse cursor over it, and "View thread" link is on white background when hover.

**Comment by kivan-mih** [comment]
Hmmm... since today all background became white? Does anyone has this problem?

**Comment by seanpianka** [comment]
@laurencee Applying the original changes to the "new version folder"/index.js and/or ssb-interop.js does not seem to function properly (the background of the chat is still white w/ white text).

**Comment by laurencee** [comment]
I just started getting this too with the white background, will have to check what changed. Guess my update was delayed.

**Comment by quietconundrum** [comment]
can confirm on mac, above overwrites suggested for index.js and ssb-interop.js have no effect after latest update. white background has returned once again.

**Comment by laurencee** [comment]
Well the above fixes were just mean to fix a few different styles that had been introduced/renamed. The main script is still https://github.com/widget-/slack-black-theme/blob/master/custom.css and that hasn't been updated since the repo owner disappeared.

**Comment by Arezhik** [comment]
Not perfect, but here is a repo for the most recent update change. Feel free to add issues for me to fix.

https://github.com/Arezhik/slackCustomCSS

**Comment by wickkidd** [comment]
v3.3.1 breaks the message window styling. The sidebar styling still works.

re: https://github.com/widget-/slack-black-theme/issues/39#issuecomment-356698404
